feflow/utils/lambda_protocol.py

```diff
@@ -0,0 +1,334 @@
+# License: MIT
+
+import numpy as np
+import warnings
+import copy
+from openmmtools.alchemy import AlchemicalState
+
+
+class LambdaProtocol:
+    """Protocols for perturbing each of the component energy terms in alchemical
+    free energy simulations.
+
+    TODO
+    ----
+    * Class needs cleaning up and made more consistent
+    """
+
+    default_functions = {'lambda_sterics_core':
+                         lambda x: x,
+                         'lambda_electrostatics_core':
+                         lambda x: x,
+                         'lambda_sterics_insert':
+                         lambda x: 2.0 * x if x < 0.5 else 1.0,
+                         'lambda_sterics_delete':
+                         lambda x: 0.0 if x < 0.5 else 2.0 * (x - 0.5),
+                         'lambda_electrostatics_insert':
+                         lambda x: 0.0 if x < 0.5 else 2.0 * (x - 0.5),
+                         'lambda_electrostatics_delete':
+                         lambda x: 2.0 * x if x < 0.5 else 1.0,
+                         'lambda_bonds':
+                         lambda x: x,
+                         'lambda_angles':
+                         lambda x: x,
+                         'lambda_torsions':
+                         lambda x: x
+                         }
+
+    # lambda components for each component,
+    # all run from 0 -> 1 following master lambda
+    def __init__(self, functions='default', windows=10, lambda_schedule=None):
```

Comment:
Just a note here that this class has a different behaviour to Perses' initial version, primarily because we deal with some lambda validation elsewhere in the Protocols.